_posts/2017-06-01-how-to-build-dotnet-core-project-with-vs-code.md

```diff
@@ -0,0 +1,375 @@
+* [C# Extensions](https://marketplace.visualstudio.com/items?itemName=jchannon.csharpextensions) - Add some snippets (ctor, prop, ...) and the template for a Class or an Interface.
+
+I advise to have at least C# By OmniSharp & Nuget Package Manger, because without them, it could be harder to work on a .net core project with VS Code... But it still possible ! 
+
+### <a name='DebuggingwithVisualCode'></a>Debugging with Visual Code
+
+By default, on the left side of you VSCode, you have a crossed bug representing the Debug panel. On this panel, there is everything you need to debug a .net core project : 
+* Variables contains current values of your variables when the debugger hit a breakpoint
+* Watch allow the developer to have a look on specific variable and change its value
+* CallStack redraw the path of your execution
+
+If you hit F5 or you press the play button, VSCode will ask you to configure the project : 
+![vscode hit F5](/assets/images/blog/201706-dotnet-core-vscode/vscode-hit-f5.png)
+
+Because I have installed C# By OmniSharp, there is .NET Core in the list of environment available. Select .NET Core and a launch.json file will be create in .vscode folder. There is three nodes on ``configurations`` : 
+* **.NET Core Launch (console)** to run your project as console.
+* **.NET Core Launch (web)** to run your project as web application.
+* **.NET Core Attach** to attach an instance of your application on the debugger.
+
+This file is used by VSCode to configure your debugger. There is IntelliSense to add few properties  and you can found more documentation [here](https://code.visualstudio.com/docs/editor/debugging).
+
+On my configuration, I update the **program** property to set the StartUp project from ``"program": "${workspaceRoot}/bin/Debug/<target-framework>/<project-name.dll>",`` to ``"program": "${workspaceRoot}/dotnetcorevscode.Web/bin/Debug/netcoreapp1.1/dotnetcorevscode.Web.dll",``.
+
+If I hit F5 one more time, I'll have an Error : "Could not find the preLaunchTask 'build'." You can create this on by click on "Configure Task Runner" and select .NET Core once again. It will create a launch.json file with the ``dotnet build`` event.
+
+_When you run you project you'll have an exception because appsettings.json is not found. You can update the path or move the file on the root of your project_
+
+Finally, on Debug Console : 
+
+```
+Hosting environment: Development
+Content root path: c:\Projects\dotnetcore-vscode
+Launching browser (cmd.exe /C start http://localhost:5000)
+Now listening on: http://localhost:5000
+Application started. Press Ctrl+C to shut down.
+```
+
+### <a name='UnittestingwithVisualCode'></a>Unit testing with Visual Code
+
+_N.B. : When I was writing this post, there were some difficulties to run unit test in VSCode, but the OmniSharp team works good and since the 25th May, we got a little feature on OmniSharp 1.10 : `MSTest support added`_
```

Comment:
A la place de "When I was writing this post", j'écrirais "While writing this post", je trouve la formulation moins lourde, mais elle est peut-être plus familière